src/Contract.ts

```diff
@@ -50,6 +50,8 @@ export class Contract {
       fn(...(args.args || [])),
       this.environment,
       args.from,
+      this.contract.address,
```

Comment:
This is not necessary because `contract` is already public so you could simply use `transaction.contract.address` in your code.